src/loc.ml

```diff
@@ -85,6 +103,15 @@ let print ppf loc =
           Format.fprintf pp "%s\n%*s\n" line
             stop_c
             (String.make len '^')
+        else if num_lines <= 10 && not (start = loc_file.start) then
```

Comment:
Does that work when the error starts at the beginning of the file, but spans only the first stanza for example?

Reply:
Thinking more about it i'm wondering if I should handle this special case like this. I can see the following options: 

* I'm guessing the only way to check this would be to actually verify that the stanza reported is indeed the entire file. I could check that  `stop.pos_lnum` is the last line of the file and `start` equals `loc_file.start`.

* Print the file and follow the same rules as the remaining stanzas about not displaying anything more than 10 lines.

What do you think?

Reply:
I've pushed a change removing the `start = loc_file.start` check for now. I don't mind updating the logic to whatever you or @rgrinberg feel will be more suited here.